tensorflow_addons/losses/multi_similarity_loss.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implements Multi Similarity loss."""
+
+import tensorflow as tf
+from tensorflow_addons.utils.types import FloatTensorLike, TensorLike
+from typeguard import typechecked
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class MultiSimilarityLoss(tf.keras.losses.Loss):
```

Comment:
Any reason that we do not subclass `LossFunctionWrapper`?

Reply:
Python private APIs not allowed.

Reply:
https://github.com/tensorflow/addons/blob/40d8e59cbff54204affbfc44ae3dbf8021aab427/tensorflow_addons/utils/keras_utils.py#L24

Reply:
Got it. Will transform the file. Give me sometime.